### PR TITLE
Add test to track changes to VSIX package contents

### PR DESCRIPTION
**Pull request by drewnoakes**
At some point in recent history, our ProjectSystem VSIX picked up a bunch of unexpected files (see #6612). It took us a while to notice this, at which point it was unclear which change caused this to regress.

This commit adds the current list of files as a text file that is under version control. There is one file per VSIX (we have three). Any addition or removal from a VSIX will cause a unit test to fail, which will draw our attention to the change immediately. If the change is intentional, we need only update a text file. If not, we can investigate immediately and block merging a regression.

This PR introduces a dependency on the `Verify.Xunit` library, which is the xUnit flavour of the Verify project:

https://github.com/VerifyTests/Verify

> Verify is a snapshot tool that simplifies the assertion of complex data models and documents.
>
> Verify is called on the test result during the assertion phase. It serializes that result and stores it in a file that matches the test name. On the next test execution, the result is again serialized and compared to the existing file. The test will fail if the two snapshots do not match: either the change is unexpected, or the reference snapshot needs to be updated to the new result.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/7805)

### CODE REVIEW

File: tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/Setup/PackageContentTests.ProjectSystem.verified.txt [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+  ko/Microsoft.VisualStudio.ProjectSystem.Managed.VS.resources.dll,
+  pl/Microsoft.VisualStudio.ProjectSystem.Managed.VS.resources.dll,
+  pt-BR/Microsoft.VisualStudio.ProjectSystem.Managed.VS.resources.dll,
+  ru/Microsoft.VisualStudio.ProjectSystem.Managed.VS.resources.dll,
+  tr/Microsoft.VisualStudio.ProjectSystem.Managed.VS.resources.dll,
+  zh-Hans/Microsoft.VisualStudio.ProjectSystem.Managed.VS.resources.dll,
+  zh-Hant/Microsoft.VisualStudio.ProjectSystem.Managed.VS.resources.dll,
+  Microsoft.VisualStudio.ProjectSystem.Managed.VS.pkgdef,
+  Microsoft.VisualStudio.ProjectSystem.Managed.dll,
+  ProjectSystem/Rules/Dependencies/AnalyzerReference.cs.xaml,
```
Comment (drewnoakes) [inline tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/Setup/PackageContentTests.ProjectSystem.verified.txt:22]: The `ProjectSystem` folder should not be present (#6612).

File: tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/Setup/PackageContentTests.ProjectSystem.verified.txt [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+  ProjectSystem/Rules/Items/ProjectItemsSchema.FSharp.it.xaml,
+  ProjectSystem/Rules/Items/ProjectItemsSchema.FSharp.ja.xaml,
+  ProjectSystem/Rules/Items/ProjectItemsSchema.FSharp.ko.xaml,
+  ProjectSystem/Rules/Items/ProjectItemsSchema.FSharp.pl.xaml,
+  ProjectSystem/Rules/Items/ProjectItemsSchema.FSharp.pt-BR.xaml,
+  ProjectSystem/Rules/Items/ProjectItemsSchema.FSharp.ru.xaml,
+  ProjectSystem/Rules/Items/ProjectItemsSchema.FSharp.tr.xaml,
+  ProjectSystem/Rules/Items/ProjectItemsSchema.FSharp.zh-Hans.xaml,
+  ProjectSystem/Rules/Items/ProjectItemsSchema.FSharp.zh-Hant.xaml,
+  AnalyzerReference.cs,
```
Comment (drewnoakes) [inline tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/Setup/PackageContentTests.ProjectSystem.verified.txt:490]: None of these `.cs` files should be present (#6612).

**Comment by drewnoakes** [comment]
The package `Verify.Xunit` is not available in any of our feeds. Investigating.

**Comment by drewnoakes** [comment]
> ProjectSystem/Rules/Dependencies/AnalyzerReference.cs.xaml

A lower version is available. I've updated the PR to use that instead.

**Review by tmeschter — APPROVED** [review]
Could you add a description of what the Verify.Xunit package does to the PR description? I'm not familiar with it.

**Pull request merged**